### ISSUE TITLE
Naming of affected and changed EObjects

### ISSUE THREAD

**Issue by HeikoKlare** [vitruv-tools/Vitruv#333]
Currently, the term `affectedEObject` is used for an objects that are either changed (in terms of an attribute or reference change) or created/deleted by a change.
According to the discussion in #331, we have to consider two improvements:
1. `VitruviusChange` contains the `getAffectedEObjects` method, which does actually _not_ only return affected objects, but also the objects referenced in changes. This is not intuitive and the method should definitely be renamed.
2. `affectedEObjects` may not be comprehensive term, so we could think about adapting that term in general. The proposal of renaming it to `changedEObjects` is not sufficient, because `changedEObjects` only cover those whose attributes or references were changed, but not the ones that were created or deleted. One option is to check whether we actually need to combine changed, created and deleted objects or if it would also be find to separate them. Another option would be to find a different term for `affected` that is clearer than it currently is. Maybe, however, the first improvement is already sufficient.

While the first improvement is essential and should be a quick improvement, the second requires a large amount of changes, so we should be sure how to improve before starting such a refactoring.

Feel free to make proposals, post your opinion and start a discussion.

**Comment by jGleitz** [comment]
> `changedEObjects` only cover those whose attributes or references were changed, but not the ones that were created or deleted

It seems that it is still not clear what the method `getChangedEObjects` or `getAffectedEObjects` should actually return. If an object `B` was inserted into `A`, I’d say that only `A` is a `changed` object. `B` is only an `involved` object (or something like that).

The same goes for deleted objects: Only the ex-container is changed, the removed object is only `involved`.

Using this logic means that there are changes without changed objects (root insertions / deletions), but I don’t think that this is a problem.

**Comment by HeikoKlare** [comment]
I think creation/deletion and insertion/removal is mixed up here. Insertion and removal are about putting some object into the containment reference of some other object, or into the contents of a resource. Creations and deletion are about the instantiation of an object and are not related to the insertion and removal of them.
So in addition to root insertions and removals, also element creations and deletions do not have a `changed` object.
For the initial problem of `changedResource` identification, it is also important that root insertions and removals do not have a `changed` object, because this results in a change resource not being identified by only considering the `changed` objects.

**Comment by jGleitz** [comment]
Ah, we do have changes for an object to be created or deleted? I thought I remembered distinctly that Vitruv does not have that, but purposefully only models insertion / removal.

**Comment by HeikoKlare** [comment]
Yes, we have these types of changes. You even need them, because without them the change model is insufficient.
